### ISSUE TITLE
Crackling Audio on Windows

### ISSUE THREAD

**Issue by thestr4ng3r** [thestr4ng3r/chiaki#1]
I already tried to play with the `bufferSize` of `QAudioSource`, but it didn't seem to help.

**Comment by die4teoffenbarung** [comment]
I don't know if this is helpful for you but i give it a try. :P
https://github.com/moonlight-stream/moonlight-qt/issues/152

**Comment by Scrxtchy** [comment]
The general implementation seems to be fine, it can play fine on some hardware (eg my speakers), and crackle on others (my headphones)
Both are running at 32bit 48000hz, however my headphones run at 5.1 Surround, not stereo, unsure if this changes anything for the QAudioOutput

**Comment by Whovian9369** [comment]
Also confirming this on Windows 10 with my stereo headphones -- Audio in other apps doesn't crackle like Chiaki does.

**Comment by Scrxtchy** [comment]
I spent a while trying to chase it down, the connection is fine and I'm sure it's something misconfigured with the qt shit, but I also don't just have the time to setup the dev tools to build this in my own time.

If I had experience with QT I could figure out how to probably change the windows audio api (MME/wasapi), but I don't. It could help. Windows Audio Session is more modern, but I've had some problems with it on certain hardware, figuring this out and letting users pick the option to switch to the older api will help if api change fixes the issues

**Comment by vua59335** [comment]
same

**Comment by Nigh7Stalk3r** [comment]
Also having this issue on Windows 10 with the latest build.

**Comment by Scrxtchy** [comment]
I don't think we need more comments of people having the issue, we somewhat know what's causing it. We'll only really need extra feedback when time goes into developing a patch

**Comment by DannyDelusion** [comment]
I've tried switching MME/WASAPI/WDM and still have a crackling issue, Win7.
Buffer size does nothing for it, the crackling is still present and audible, even through various buffer rates.
i'm wondering if it's an issue with OpusDecoder and callback time/Buffer Duration?

**Comment by Ofikabacon** [comment]
I was messing around with the files and managed to find a fix that worked for me, I simply went into the audio folder and removed `qtaudio_windows.dll` and it fixed the issue on Windows 10. I looked up the issue and came across an old steam thread talking about the same issue [here](https://steamcommunity.com/app/390540/discussions/1/1843493219427816703/) which says for Windows 7 to instead remove the `qtaudio_wasapi.dll` file. 

I managed to lower my buffer size to 1024 without any crackling, so I guess something is causing it to select the wrong audio driver or they are conflicting somehow.

**Comment by Scrxtchy** [comment]
I thought this sounded bullshit reading that it was fixed by playing with compiled files but to my surprise it actually worked. Granted no audio comes out of my headphones from chiaki at all anymore. But I was able to replicate the crackle pre-fix onto some Rift speakers. Removing `qt_windows.dll` fixed the issue for that device

**Comment by Ofikabacon** [comment]
Yea I was surprised that it worked that's why I looked up the issue, I did some more testing and I could only get 48k sample rate to work. I am using a scarlett solo interface which has 2 channels and I checked 44.1k at 16 and 24 bit and 96k also at 16 and 24 bit, both are silent. I also tried changing my interface's buffer size but that didn't work either.

**Comment by Scrxtchy** [comment]
This might be where we need to look
https://doc.qt.io/qt-5/qaudiosystemplugin.html
Choosing between wasapi and WinMM could be key. As per the High Fidelity developer has noted, WinMM seems to be choice for Win7 users

**Comment by TheManchineel** [comment]
Setting sample rate to 48 KHz and renaming qtaudio_windows.dll removed the crackling noise, but now the sound feels quite delayed.

**Comment by auto-mation** [comment]
> Setting sample rate to 48 KHz and renaming qtaudio_windows.dll removed the crackling noise, but now the sound feels quite delayed.

if you put in buffer something very low like 2400 the delay is almost 0, but it sounds horrible, not entirely sure what's the issue

**Comment by sverbist** [comment]
I have crackling audio as well, but I'm running Linux. Should I create a new issue?

**Comment by Scrxtchy** [comment]
I would suggest doing so to keep the issues on topic per OS

**Comment by DustinHanke** [comment]
> Setting sample rate to 48 KHz and renaming qtaudio_windows.dll removed the crackling noise, but now the sound feels quite delayed.

It worked for me :) Using 4000 buffer size right now and it isn't that delayed as well.

**Comment by DannyDelusion** [comment]
> 
> 
> > Setting sample rate to 48 KHz and renaming qtaudio_windows.dll removed the crackling noise, but now the sound feels quite delayed.
> 
> It worked for me :) Using 4000 buffer size right now and it isn't that delayed as well.

Win 7 - still not working as a fix for me, i think win7 just needs to wait for a WinMM/Wasapi fix

**Comment by Flying-koala** [comment]
Win 10 - When changing the output for the app in "App volume and device preferences" it will always go with the default device, no matter what is chosen. Could that be related to this problem?

**Comment by leonardobortolotti** [comment]
I tried delete the `qtaudio_windows.dll` and then setting  my headphone as the main device for Chiaki in Windows Settings > App Volume and device preferences, and it worked.
chiaki 1.2.1 , Windows 10

**Comment by Flying-koala** [comment]
I tried delete the qtaudio_windows.dll before to fix the crackling with no luck, but this time it work. So that's good. 
For me, Chiaki will use the default device that was set when it was opened. I do have other audio issues (unrelated), so someone should double check that.

**Comment by IrealiTY** [comment]
Can confirm what @Flying-koala said. Deleting the `qtaudio_windows.dll` and forcing the Audio settings in Windows (screenshot attached) seems to fix the audio crackling ....
![image](https://user-images.githubusercontent.com/1223686/87951262-a6348e00-caa8-11ea-8ddb-5e06cf518b35.png)

BUT at least on my side, it creates a HUGE audio delay (2-3seconds?) BUT² lowering the Audio Buffer to 1024 seems to do the trick?
![image](https://user-images.githubusercontent.com/1223686/87951170-88672900-caa8-11ea-9fd5-0cc7dd711c59.png)

**Comment by barisyild** [comment]
i am having exactly the same issue, @Flying-koala fix works.

**Comment by b4zyl** [comment]
Does't work for me :( when i delete qtaudio_windows.dll there is no audio at all (chiaki 1.3.0) and i cannot force anything in Windows Sound setup.

**Comment by b4zyl** [comment]
@Flying-koala fix worked only once for me. Now, when i set default sound device for Chiaki and remove qtaudio_windows.dll i don't have sound at all. Restore file to the audio folder gives crackling sound again.

**Comment by jasonpremchand** [comment]
@Flying-koala 
Zipped a copy of qtaudio_windows.dll as backup
Deleted the qtaudio_windows.dll
Set audio bitrate to 7200

No blotchy/stuttering/skipping audio.

Thank you!
